packages/block-editor/src/layouts/flow.js

```diff
@@ -117,6 +117,9 @@ export default {
 				? `
 					${ appendSelectors( selector, '> *' ) } {
 						max-width: ${ contentSize ?? wideSize };
+					}
+
+					${ appendSelectors( selector, '> :not(.alignleft):not(.alignright)' ) } {
```

Comment:
I changed this that way the auto margins don't apply to left/right aligned blocks. Otherwise, the margins we had specifically for these blocks didn't apply at all. I think this has some small potential to impact FSE themes for aligned left/right blocks but this restores the original intent basically. I think it's fine but also curious about testing impact here.

Reply:
I guess also the "max-width" rule should be moved here maybe, let me know what you think?

Reply:
To be sure I understand right, the desired behavior is that a floated image is omitted from max width and auto margins, so that it's up to a theme to decide whether it should be [part of the main column](https://codepen.io/joen/pen/zLWvrW), or not. Is that right?

If so, then that appears to work fairly well in this branch:

<img width="1631" alt="Screenshot 2022-02-10 at 12 13 07" src="https://user-images.githubusercontent.com/1204802/153396217-03c0b932-c8a2-43a2-a2fb-9e55275ef98a.png">

<img width="2032" alt="Screenshot 2022-02-10 at 12 13 47" src="https://user-images.githubusercontent.com/1204802/153396235-d94a3be7-eebf-4248-9514-f7f7f89fea62.png">

However, look at the first paragraph in that screenshot above, it no longer has a max-width for some reason, whereas it does in trunk:

<img width="2032" alt="Screenshot 2022-02-10 at 12 14 37" src="https://user-images.githubusercontent.com/1204802/153396327-7d03f581-9769-4438-9363-d7097db03a5d.png">

Based on later comments in Maggie's testing content, it sounds like this is could be correct:

<img width="1437" alt="Screenshot 2022-02-10 at 12 17 09" src="https://user-images.githubusercontent.com/1204802/153396524-e13c5f83-a2fd-46a0-9d60-a4660b815d61.png">
<img width="1425" alt="Screenshot 2022-02-10 at 12 17 15" src="https://user-images.githubusercontent.com/1204802/153396567-b64685ae-bc96-4745-af03-86c6c93b6c77.png">

However it is inconsistent with the editor 🤔 :
<img width="1407" alt="Screenshot 2022-02-10 at 12 17 43" src="https://user-images.githubusercontent.com/1204802/153396589-816c6eff-3633-43d5-92c7-091f328a4c64.png">

Reply:
> To be sure I understand right, the desired behavior is that a floated image is omitted from max width and auto margins, so that it's up to a theme to decide whether it should be [part of the main column](https://codepen.io/joen/pen/zLWvrW), or not. Is that right?

Yes I want to exclude left/right aligned blocks from the auto margins and the max width. But it's not up to the theme to decide whether it should be part of the main column. It can't be part of the main column at all. If you want something to be part of the main column, you need to wrap it in a group.

Reply:
My only concern here is whether this added specificity is fine. I think it is but I'm not sure.

Reply:
Yep, sounds good to me. Did you see the Paragraph missing a max-width, issue, though? That's present in this branch but not in trunk.

Reply:
yes, it was actually a missing `.` it should be fixed in the last commit.

Reply:
Good catch. 

I wonder if this would work? `:where(:not(.alignleft):not(.alignright))`

Reply:
@jasmussen's suggestion solves the alignment issue for me